doc/source/ray-core/miscellaneous.rst

```diff
@@ -216,3 +216,135 @@ To get information about the current available resource capacity of your cluster
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Because all nodes and workers connect to the GCS, many network connections will
+be created and the operating system has to support that number of connections.
+
+Maximum open files
+******************
+
+The OS has to be configured to support opening many TCP connections since every
+worker and raylet connects to the GCS. In POSIX systems, the current limit can
+be checked by `ulimit -n` and if it's small, it should be increased according to
+the OS manual. 
+
+ARP cache
+*********
+
+Another thing that needs to be configured is the ARP cache. In a large cluster,
+all the worker nodes connect to the head node, which adds a lot of entries to
+the ARP table. Ensure that the ARP cache size is large enough to handle this
+many nodes.
+Failure to do this will result in the head node hanging. When this happens,
+`dmesg` will show errors like `neighbor table overflow message`.
+
+In Ubuntu, the ARP cache size can be tuned in `/etc/sysctl.conf` by increasing
+the value of `net.ipv4.neigh.default.gc_thresh1` - `net.ipv4.neigh.default.gc_thresh3`. 
+For more details, please refer to the OS manual.
+
+Tuning Ray Settings
+~~~~~~~~~~~~~~~~~~~
+
+To run a large cluster, several parameters need to be tuned in Ray.
+
+Health check
+************
+
+GCS checks the health of the worker nodes by sending ping periodically.
+Sometimes, if the network is poor or the raylet is too busy to respond,
+the raylet might be marked dead incorrectly. 
+
+GCS uses a similar algorithm as k8s probes to check the healthiness of the
+Raylet: it sends the ping to the worker periodically and if it failed X times
+consecutively, the worker will be marked as dead. Once a worker is marked as
+dead, it'll be considered dead forever even it's not. The following OS
+environments can be used to tune this: 
+
+- `RAY_health_check_initial_delay_ms` The delay of the first health check, 5000ms by default.
+- `RAY_health_check_period_ms` The interval between two health check requests, 3000ms by default.
+- `RAY_health_check_timeout_ms` The timeout to consider a health check failed, 10000ms by default.
+- `RAY_health_check_failure_threshold` The consecutive failure threshold for the
+  worker node being considered dead, 5 by default. 
+
+
+Resource broadcasting
+*********************
+
+Another functionality GCS provided is to ensure each worker node has a view of
+available resources of each other in the Ray cluster. Each raylet is going to
+push its local available resource to GCS and GCS will broadcast it to all the
+raylet periodically. The time complexity is O(N^2). In a large Ray cluster, this
+is going to be an issue, since most of the time is spent on broadcasting the
+resources. There are several settings we can use to tune this: 
+
+- `RAY_resource_broadcast_batch_size` The maximum number of nodes in a single
+  request sent by GCS, by default 512.
+- `RAY_raylet_report_resources_period_milliseconds` The interval between two
```

Comment:
Same question here, I thought this is no longer an issue with the pull model?

Reply:
For resource broadcasting, it's still push mode now. The pull based is wip. 
We need this https://github.com/ray-project/ray/pull/30460  merged and do complete tests.
The new plan is to make it experimental in 2.3 and turn it on in 2.4 by default.

I think maybe we can keep it here for now so that users who is using ray 2.2 can benefit this. Once we have that PR merged, and pass the benchmark, we can update this to enable pull based broadcasting.

After 2.4, we maybe can delete this.

I think this part is to give the user some instruction to run ray in large scale and report the unknown issues found.

Reply:
I see. Why don't we file a GitHub issue with planned scalability improvements, and link to it from the doc? That way there is some more communication about which of these will be improved/unneeded in the future.

One issue with documenting these at all is that they show up in search randomly, and users might not realize it doesn't apply to future Ray versions.